conjure-java-core/src/test/resources/example-types.yml

```diff
@@ -47,6 +47,9 @@ types:
         fields:
           item: optional<any>
           setItem: optional<set<StringAliasExample>>
+      OptionalAliasExample:
+        fields:
+          optionalAlias: OptionalAlias
```

Comment:
I'm pretty sure the `toConjureTypeWithoutAliases` already does the right thing, but maybe no harm in having a more deeply nested one to prove we're not just unwrapping the top level of aliasing? e.g. `AliasOfAliasOfOptional`?

Reply:
I'm less confident that we already generate empty constructors for nested aliases, there's definitely some room for improvement in that regard. It's a tad more difficult because we must construct an empty instance of the delegate in generated constructor, and multidimensional aliases are somewhat less common.

We could add a `public static MyAlias empty()` factory which returns a singleton. Rather than building all the nested layers of empty constructor arguments we could simply call the delegate, which in turn handles its delegate. Either way I'd prefer to use a separate PR.